### DISCUSSION TITLE
Difference between Digital Asset (Token Objects) creation functions

### DISCUSSION THREAD

**Discussion by gregnazario** [aptos-labs/aptos-developer-discussions#49] [Questions] [answered]
From Discord user SoumalyaThecoder:

> Hey can anyone tell me what's difference between mint_token_object(), &token.crete(), token::create_named_token, nft - in aptos move

**Comment by gregnazario** [comment, marked as answer]
There are two standards for Aptos NFTs:

1. Digital Assets (the new standard) - original named token_object
2. Legacy Token ( the old standard)

For the Digital asset standard there are a few ways to create tokens (with code):
1. `0x4::token::create` - this creates an object with a randomized address.  This NFT can be deleted to get storage cost back https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/framework/aptos-token-objects/sources/token.move#L207-L221
2. `0x4::token::create_numbered_token` - this creates a token where each one is numbered in order, while staying fully parallel in minting (high throughput).  This NFT can be deleted to get storage cost back https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/framework/aptos-token-objects/sources/token.move#L223C1-L241C6
3. `0x4::token::create_named_token` - this creates a token using the name of the token as a seed to the address of the object.  This makes it possible to derive the address on-chain without use of an indexer, but has restrictions.  Two tokens can't have the same name, if the token changes names, it could be hard to find, and the NFT cannot be deleted to get storage cost back. https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/framework/aptos-token-objects/sources/token.move#L243-L259

**Comment by soumalya340** [comment]
Hey @gregnazario  im bit confused just , a small question i have  
that why do we need `create_fixed_collection` object -  
```rust
collection::create_fixed_collection(
            &resource_signer,
            string::utf8(COLLECTION_DESCRIPTION),
            SUPPLY,
            string::utf8(COLLECTION_NAME),
            option::some(royalty),
            string::utf8(COLLECTION_URI)
        ); , 
```
how does it involved with `create_named_token`

**Reply by 0x-j** [reply]
The usual flow for an NFT collection is you first create a collection, then create token (i.e. NFT) inside that collection. Take the Aptogotchi NFT collection as an example, you first [create the NFT collection at init](https://github.com/aptos-labs/aptogotchi/blob/main/move/sources/aptogotchi.move#L110). Then [create a token when someone creates an Aptogotchi NFT](https://github.com/aptos-labs/aptogotchi/blob/main/move/sources/aptogotchi.move#L149). 

Token creation can be done by calling any of the [`mint_token_object`](https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/framework/aptos-token-objects/sources/aptos_token.move#L183) or `create` or `create_numbered_token` or `create_named_token`. They come with different trade off as Greg mentioned above.

**Reply by gregnazario** [reply]
A digital asset collection has 2 pieces to it.  A collection object (describes the details around the collection) and handles the relationship between tokens.

The token objects, then refer to the collection object.  This allows for parallelism, and data separation for efficiency of storage.

![object_model](https://github.com/aptos-labs/aptos-developer-discussions/assets/1335270/47913a01-6db6-47cb-bd37-d1bda3f1d922)
